### PR TITLE
Minor: Improve Type documentation

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?
N/A

# Rationale for this change
 
Inspired by @tshauck 's experience in https://github.com/apache/arrow-rs/pull/6199 and  @tustvold 's comment: https://github.com/apache/arrow-rs/pull/6199#issuecomment-2275579948

# What changes are included in this PR?

1. Add a note to `Date64Type` about representation restriction and link to `DataType::Date64`
2. Drive by cleanup to make other data type doc strings more concise

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-array/src/types.rs [hunk 1]
```diff
@@ -101,163 +101,165 @@ make_type!(
...
+    "32-bit date type: the elapsed time since UNIX epoch in days (32 bits)."
 );
 make_type!(
     Date64Type,
     i64,
     DataType::Date64,
-    "A 64-bit date type representing the elapsed time since UNIX epoch in milliseconds(64 bits)."
+    "64-bit date type: the elapsed time since UNIX epoch in milliseconds (64 bits). \
+    Values must be divisible by `86_400_000`. \
+    See [DataType::Date64] for more details."
```
Comment (crepererum) [inline arrow-array/src/types.rs:196]:
```suggestion
    See [`DataType::Date64`] for more details."
```

Comment (crepererum) [reply arrow-array/src/types.rs:196]: I'm also not entirely sure if this newline handling is correct in markdown. Might be that this is

```
foo
    bar
    baz
```

in markdown, which might result in a code block (depending on the parser).

**Review by crepererum — APPROVED** [review]

File: arrow-array/src/types.rs (hunk 1 above)
Comment (alamb) [reply arrow-array/src/types.rs:196]: Here is how it renders locally. 

Screenshot 2024-08-13 at 11 54 48 AM

I could not figure out how to get a newline into the documentation

**Comment by alamb** [comment]
Thanks for the review @crepererum

**Pull request merged**